sled-hardware/src/disk.rs

```diff
@@ -179,18 +183,93 @@ pub struct Disk {
     zpool_name: ZpoolName,
 }
 
+pub const FACTORY_DATASET: &'static str = "factory";
+pub const INSTALL_DATASET: &'static str = "install";
+pub const CRASH_DATASET: &'static str = "crash";
+pub const CLUSTER_DATASET: &'static str = "cluster";
+pub const CONFIG_DATASET: &'static str = "config";
+pub const ZONE_DATASET: &'static str = "zone";
+
+const U2_EXPECTED_DATASET_COUNT: usize = 1;
+static U2_EXPECTED_DATASETS: [&'static str; U2_EXPECTED_DATASET_COUNT] = [
+    // Stores filesystems for zones
+    ZONE_DATASET,
+];
+
+const M2_EXPECTED_DATASET_COUNT: usize = 5;
+static M2_EXPECTED_DATASETS: [&'static str; M2_EXPECTED_DATASET_COUNT] = [
+    // Stores a "factory install" set of software
+    FACTORY_DATASET,
+    // Stores software images.
+    //
+    // Should be duplicated to both M.2s.
+    INSTALL_DATASET,
```

Comment:
This gives us a landing spot for the update process - we should be placing non-ramdisk control plane zones within

```
/pool/int/<UUID>/install
```

On each of the M.2s